cfme/tests/automate/custom_button/test_service_objects.py

```diff
@@ -484,7 +490,62 @@ def test_custom_button_dialog_service_archived():
     Bugzilla:
         1439883
     """
-    pass
+    service, vm = service_vm
+    group, obj_type = button_group
+
+    with appliance.context.use(ViaUI):
+        button = group.buttons.create(
+            text=fauxfactory.gen_alphanumeric(start="btn_"),
+            hover=fauxfactory.gen_alphanumeric(start="hover_"),
+            dialog=dialog,
+            system="Request",
+            request="InspectMe",
+        )
+    request.addfinalizer(button.delete_if_exists)
+
+    for with_vm in [True, False]:  # [vm, archive_vm]
```

Comment:
This seems like a parameter to me -- why have you decided not to parametrize?

Reply:
The test was simple means to check button execution on `service` having vm in `archive` state... but just on loop give me more checks :D.